docs/05_operations/02_commands/04_Genx_CLI.md

```diff
@@ -0,0 +1,448 @@
+id: genx-commands
+keywords: [operations, genx, commands]
+tags:
+    - genx
+    - cli
+    - commands
+    - terminal
+---
+<!-- Genx  -->
+
+Genx CLI is a tool that simplifies and speeds up local development. Among other useful functions, it enables you to:
+
+- Scaffold Genesis applications from scratch based on existing seeds;
+- Start local development server;
+- Build artifacts for production;
+- Execute unit and end-to-end tests and collect coverage data;
+- Perform code linting.
+
+The `genx` command has several options available to create, configure or modify a project. these are the available parameters:
+
+|Command | Argument      | Description|
+|--------|---------------|-------------|
+| [init](#init)   |`<app_Name>`          |   create a new project |
+| [analyze](#analyze) |[folder]       |analyse the production bundle |
+| [clean](#clean)   |[...paths]     | delete specified paths (defaults to dist folder) |
+| [build](#build)  |[folder]       | build a production bundle |
+| [dev](#dev)    |[folder]       |   start a development server |
+| [run](#run)    |`<task>` [module]|  run monorepo task|
+| [serve](#serve)  |[folder]       | start a static file server (defaults to dist folder)|
+| [test](#test)   |[folder]       | run tests|
+| [lint](#lint)   |[folder]       | Lint TypesSript and style files |
+| [upgrade](#upgrade) | [folder]      | upgrade Foundation UI dependencies |
+
+## How to use it
+
+There are two main ways to run the Genx CLI:
+
+- Standalone - outside of an existing project such as when scaffolding a new project
+- Local - within an existing project such as when executing builds within a project
+
+### Standalone
+
+```terminal
+npx -y @genesislcap/genx@latest <command> [args]
+```
+
+For example, to scaffold a new project:
+
+```terminal
+npx -y @genesislcap/genx@latest init myApp
+```
+
+This will use the latest available version of the CLI and crete a new application called **myApp** usind the default seed **blank-app-seed**.
+
+### Local
+
+Once you have a project, you can add Genx CLI as a local dependency to your **package.json**:
+
+``` javascript
+"devDependencies": {
+    ...
+    "@genesislcap/genx": "...",
+    ...
+},
+```
+
+You can find out the latest available version by running `npm info @genesislcap/genx` at any point.
+
+Once you have bootstrapped NPM dependencies, you will be able to call Genx CLI in **package.json** scripts section like so:
+
+``` javascript
+"scripts":{
+    ...
+    "build": "genx <command> [args]"
+    ...
+}
+```
+
+For example:
+
+``` javascript
+"scripts":{
+    ...
+    "build": "genx build -e ENABLE_SSO",
+    "dev": "genx dev -e API_HOST,ENABLE_SSO"
+    "serve": "genx serve",
+    "test": "genx test"
+    ...
+}
+```
+
+now if you execute `npm run build` or `npm run dev`, it will launch Genx CLI installed in your project locally.
+
+:::note
+If you are going to use a specific version, please do change the `latest` with the specific version
+:::
```

Comment:
This note is confusing - in local mode you put the specific version you want to use into package.json (you can find out the version number by `npm info @genesislcap/genx`). Then when executing just do `genx <command>` (without `@latest`).